### PR TITLE
Update grant claim notification UI

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/17133

Server-side PR: https://github.com/brave-intl/bat-go/pull/1175

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Example screenshots:

Screen Shot 2022-01-24 at 12 20 24 PM
Screen Shot 2022-01-24 at 12 20 40 PM
Screen Shot 2022-01-24 at 12 21 03 PM

### Base scenario

* Given that a user has an Ad grant available to claim
* When the user visits the newtab page, the rewards panel, or the rewards page
* Then the user should see the updated UI (see screenshots).
* And the "days to claim" value should decrease with each day.

### Upgrade scenario

* Given that a user in a previous version has an Ad grant available to claim
* When the user upgrades to this version
* Then the user should see the updated UI

Note: in the upgrade case, the browser might not have the data to display the countdown text. In that case, the countdown text will not be displayed.

### CODE REVIEW

**Review by emerick — APPROVED** [review]
LGTM, just one minor nit

**Review by szilardszaloki — APPROVED** [review]

**Comment by emerick** [comment]
Looks good on Android!

**Review by emerick — APPROVED** [review]
LGTM

File: vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/get_available/get_available.cc [hunk 1]
```diff
@@ -172,6 +172,14 @@ type::Result GetAvailable::ParseBody(
       promotion->expires_at = time.ToDoubleT();
     }
 
+    auto* claimable_until = item.FindStringKey("claimableUntil");
```
Comment (szilardszaloki) [inline vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/get_available/get_available.cc:175]: Would you mind adjusting the response body description in `get_available.h` and linking to the corresponding server side issue?

Comment (zenparsing) [reply vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/get_available/get_available.cc:175]: Thanks for reminding me about the docs in the header file. I've added the field there, but I'd prefer not to add issue links unless it's a TODO thing.

Comment (zenparsing) [reply vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/promotion/get_available/get_available.cc:175]: Ah, that makes sense :) Done!

**Pull request merged**